core/src/main/scala/org/http4s/Query.scala

```diff
@@ -23,29 +23,37 @@ import scala.collection.immutable
   * When rendered, the resulting `String` will have the pairs separated
   * by '&' while the key is separated from the value with '='
   */
-final class Query private (val pairs: Vector[KeyValue]) extends QueryOps with Renderable {
+final class Query private (value: Either[Vector[KeyValue], String])
```

Comment:
```
[error] core: Failed binary compatibility check against org.http4s:http4s-core_2.13:0.21.1! Found 1 potential problems
[error]  * method this(scala.collection.immutable.Vector)Unit in class org.http4s.Query's type is different in current version, where it is (scala.util.Either)Unit instead of (scala.collection.immutable.Vector)Unit
[error]    filter with: ProblemFilters.exclude[IncompatibleMethTypeProblem]("org.http4s.Query.this")
```

We could do it with an auxiliary constructor.  Or grant an exception, because it was private.

Reply:
fixed with auxillary constructor. Finally got a use for that language feature :p

Reply:
Yeah, I always prefer to fix it with code rather than exemptions.